Portage/GHCCore.hs

```diff
@@ -167,6 +170,42 @@ ghc843 = (ghc [8,4,3], mkIndex ghc843_pkgs)
 --      example: https://downloads.haskell.org/~ghc/8.6.5/docs/html/users_guide/8.6.5-notes.html
 --  * our binary tarballs (package.conf.d.initial subdir)
 --  * ancient: http://haskell.org/haskellwiki/Libraries_released_with_GHC
+ghc901_pkgs :: [Cabal.PackageIdentifier]
+ghc901_pkgs =
+--  , p "Cabal" [3,2,0,0]  package is upgradeable
+  [ p "array" [0,5,4,0]
+  , p "base" [4,15,0,0]
+  , p "binary" [0,8,8,0] -- used by libghc
+  , p "bytestring" [0,10,12,1]
+  , p "containers" [0,6,4,1]
+  , p "deepseq" [1,4,5,0] -- used by time
+  , p "directory" [1,3,6,1]
+--  , p "exceptions [0,10,4] -- used by haskeline, package is upgradeable 
```

Comment:
Looks like `exceptions` is used by `ghc` library itself. That effectively makes it unupgradeable.